packages/modules/pricing/src/services/pricing-module.ts

```diff
@@ -515,6 +515,11 @@ export default class PricingModuleService
     // We can make the `insert` inside upsertWithReplace do an `upsert` instead to avoid this
     const normalizedData = await this.normalizeUpdateData(data)
 
+    const priceListPrices = await this.priceService_.list({
+      price_set_id: normalizedData.map(({ id }) => id),
+      price_list_id: { $ne: null },
+    })
+
```

Comment:
thought(non-blocking): I wonder if this is the right approach to take since a price list price can potentially have a massive amount of prices so the performance of this could be bad.

It might be something to come back to later and have a more specialized endpoint to update/create individual prices without removing existing ones.